homeassistant/components/tuya/sensor.py

```diff
@@ -1450,10 +1482,7 @@ def native_value(self) -> StateType:
 
         # Scale integer/float value
         if isinstance(self._type_data, IntegerTypeData):
-            scaled_value = self._type_data.scale_value(value)
-            if self._uom and self._uom.conversion_fn is not None:
-                return self._uom.conversion_fn(scaled_value)
-            return scaled_value
+            return self._type_data.scale_value(value)
```

Comment:
This is the main change - dropping the manual conversion and relying on the base sensor platform implementation